weave/trace_server/interface/builtin_object_classes/saved_view.py

```diff
@@ -0,0 +1,89 @@
+from typing import Any, Literal, Optional
+
+from pydantic import BaseModel, Field
+
+from weave.trace.traverse import PathElement
+from weave.trace_server.interface.builtin_object_classes import base_object_def
+
+# Compare with definitions in weave/trace_server/trace_server_interface.py
+
+
+class CallsFilter(BaseModel):
+    op_names: Optional[list[str]] = None
+    input_refs: Optional[list[str]] = None
+    output_refs: Optional[list[str]] = None
+    parent_ids: Optional[list[str]] = None
+    trace_ids: Optional[list[str]] = None
+    call_ids: Optional[list[str]] = None
+    trace_roots_only: Optional[bool] = None
+    wb_user_ids: Optional[list[str]] = None
+    wb_run_ids: Optional[list[str]] = None
+
+
+class Filter(BaseModel):
+    id: int
+    field: str
+    # Type of operator could be locked down more, but this is better for extensibility
+    operator: str
+    value: Any
+
+
+class Filters(BaseModel):
+    items: list[Filter] = []
+    logicOperator: Literal["and"] = "and"
+
+
+class Pin(BaseModel):
+    # TODO: Make them optional? But one is required?
+    left: list[str]
+    right: list[str]
+
+
+SortDirection = Literal["asc", "desc"]
+
+
+# Note: This is intentionally similar to `SortBy` in `trace_server_interface.py`
+# but duplicating because logically it shouldn't need to match the API interface.
+class SortBy(BaseModel):
+    # Field should be a key of `CallSchema`. For dictionary fields
+    # (`attributes`, `inputs`, `outputs`, `summary`), the field can be
+    # dot-separated.
+    field: str
+    direction: SortDirection
+
+
+class Column(BaseModel):
+    # Optional in case we want something like computed columns in the future.
+    path: Optional[list[PathElement]] = Field(default=None)
+    label: Optional[str] = Field(default=None)
+
+
+class SavedViewDefinition(BaseModel):
+    filter: Optional[CallsFilter] = Field(default=None)
+    filters: Optional[Filters] = Field(default=None)
+
+    # cols is legacy column visibility config
```

Comment:
why do we need to include something legacy here?

Reply:
I will fix the comment, it is misleading - in the UI we are currently using a dict from field to boolean to determine column visibility. This is not ideal because it doesn't give you control over the order columns are displayed, e.g. if you wanted to display the User column before the Status you are out of luck.

I would like to move to a design where the columns shown are explicitly specified in order. The Python Saved Views implementation supports this, it will need to be added to the UI. Eventually, I want to drop the dict version.